### PR TITLE
System.Drawing: Consolidate Clip-related code in Graphics

### PR DESCRIPTION
**Pull request by qmfrederik**
- Move common Clip-related code from Graphics.Unix and Graphics.Windows to Graphics
- Move P/Invoke methods to GdiplusNative

### CODE REVIEW

File: src/System.Drawing.Common/src/System/Drawing/Graphics.Unix.cs [hunk 1]
```diff
@@ -2452,14 +2218,14 @@ public RectangleF VisibleClipBounds
...
+                var rect = new GPRECTF();
+                int status = SafeNativeMethods.Gdip.GdipGetVisibleClipBounds(new HandleRef(this, NativeGraphics), ref rect);
                 SafeNativeMethods.Gdip.CheckStatus(status);
-                return rect;
+
+                return rect.ToRectangleF();
             }
         }
-
+
```
Comment (mellinoe) [inline src/System.Drawing.Common/src/System/Drawing/Graphics.Unix.cs]: nit: remove extra whitespace

File: src/System.Drawing.Common/src/System/Drawing/Graphics.Windows.cs [hunk 2]
```diff
@@ -3434,126 +3315,23 @@ public object GetContextInfo()
...
         {
             get
             {
                 if (PrintingHelper is PrintPreviewGraphics ppGraphics)
                 {
                     return ppGraphics.VisibleClipBounds;
                 }
-
-                var rect = new GPRECTF();
+
```
Comment (mellinoe) [inline src/System.Drawing.Common/src/System/Drawing/Graphics.Windows.cs]: nit: Extra whitespace here and below.

File: src/System.Drawing.Common/src/Unix/System.Drawing/Region.cs [hunk 3]
```diff
@@ -642,6 +642,19 @@ internal IntPtr NativeObject
                 nativeRegion = value;
             }
         }
+
+        internal IntPtr _nativeRegion
```
Comment (mellinoe) [inline src/System.Drawing.Common/src/Unix/System.Drawing/Region.cs]: Are we not able to just rename the field to `_nativeRegion`?

**Comment by mellinoe** [comment]
(Also needs a rebase)

File: src/System.Drawing.Common/src/Unix/System.Drawing/Region.cs (hunk 3 above)
Comment (qmfrederik) [reply src/System.Drawing.Common/src/Unix/System.Drawing/Region.cs]: #23189 removed the Unix implementation of `Region`, so this problem fixed itself ;-)

**Comment by qmfrederik** [comment]
@mellinoe I rebased and updated the PR according to the feedback; should be good now. I'll keep an eye on the CI to see if something broke as part of the rebase, though.

File: src/System.Drawing.Common/src/System/Drawing/Graphics.Windows.cs (hunk 2 above)
Comment (mellinoe) [reply src/System.Drawing.Common/src/System/Drawing/Graphics.Windows.cs]: @qmfrederik It looks like there is still some extra whitespace in this method. Could you clean it up?

Comment (qmfrederik) [reply src/System.Drawing.Common/src/System/Drawing/Graphics.Windows.cs]: @mellinoe Oops.. hope I got it right this time.

**Comment by qmfrederik** [comment]
@dotnet-bot test Linux x64 Release Build

**Pull request merged**